pkg/liqo-controller-manager/namespaceoffloading-controller/clusterselector.go

```diff
@@ -110,3 +110,12 @@ func matchVirtualNodeSelectorTerms(ctx context.Context, cl client.Client, virtua
 	}
 	return k8shelper.MatchNodeSelectorTerms(n, selector)
 }
+
+func MatchNodeSelectorTerms(node *corev1.Node, selector *corev1.NodeSelector) (bool, error) {
+	// Shortcircuit the matching logic, to always return a positive outcome in case no selector is specified.
+	if len(selector.NodeSelectorTerms) == 0 {
+		return true, nil
+	}
+
+	return k8shelper.MatchNodeSelectorTerms(node, selector)
+}
```

Comment:
This is symptom that there is some logic based on nodes and not virtualnodes. We need to check if it necessary.

Reply:
It wasn't necessary, I switched to virtualnodes